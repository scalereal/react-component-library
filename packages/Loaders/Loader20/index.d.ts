import React from 'react';
import { WithStyle } from '../../utils';
import { Props } from '../types';
export declare const SizeMap: Props['size'][];
declare const Loader20: React.FC<Props> & WithStyle;
export default Loader20;
