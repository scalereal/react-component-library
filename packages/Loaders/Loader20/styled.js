var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
import styled from 'styled-components';
var mixin = function (size) {
    switch (size) {
        case "S":
            return "\n            height: 60px;\n            width: 60px;\n            .multi-spinner {\n                border: 3px solid transparent;\n            }\n            ";
        case "M":
            return "\n            height: 80px;\n            width: 80px;\n            .multi-spinner {\n                border: 4px solid transparent;\n            }\n            ";
        case "L":
            return "\n                height: 120px;\n                width: 120px;\n                .multi-spinner {\n                    border: 5px solid transparent;\n                }\n            ";
        case "XL":
            return "\n            height: 150px;\n            width: 150px;\n            .multi-spinner {\n                border: 5px solid transparent;\n            }\n            ";
    }
};
export var Loader = styled.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n    ", "\n\n    position: relative;\n    margin: auto;\n    overflow: hidden;\n\n    .multi-spinner {\n        position: absolute;\n        width: calc(100% - 9.9px);\n        height: calc(100% - 9.9px);\n        border-top-color: ", ";\n        border-radius: 50%;\n        -webkit-animation: spin 5s cubic-bezier(0.17, 0.49, 0.96, 0.76) infinite;\n        animation: spin 5s cubic-bezier(0.17, 0.49, 0.96, 0.76) infinite;\n    }\n\n    @-webkit-keyframes spin {\n        from {\n            -webkit-transform: rotate(0deg);\n            transform: rotate(0deg);\n        }\n        to {\n            -webkit-transform: rotate(360deg);\n            transform: rotate(360deg);\n        }\n    }\n\n    @keyframes spin {\n        from {\n            -webkit-transform: rotate(0deg);\n            transform: rotate(0deg);\n        }\n        to {\n            -webkit-transform: rotate(360deg);\n            transform: rotate(360deg);\n        }\n    }\n\n"], ["\n    ", "\n\n    position: relative;\n    margin: auto;\n    overflow: hidden;\n\n    .multi-spinner {\n        position: absolute;\n        width: calc(100% - 9.9px);\n        height: calc(100% - 9.9px);\n        border-top-color: ", ";\n        border-radius: 50%;\n        -webkit-animation: spin 5s cubic-bezier(0.17, 0.49, 0.96, 0.76) infinite;\n        animation: spin 5s cubic-bezier(0.17, 0.49, 0.96, 0.76) infinite;\n    }\n\n    @-webkit-keyframes spin {\n        from {\n            -webkit-transform: rotate(0deg);\n            transform: rotate(0deg);\n        }\n        to {\n            -webkit-transform: rotate(360deg);\n            transform: rotate(360deg);\n        }\n    }\n\n    @keyframes spin {\n        from {\n            -webkit-transform: rotate(0deg);\n            transform: rotate(0deg);\n        }\n        to {\n            -webkit-transform: rotate(360deg);\n            transform: rotate(360deg);\n        }\n    }\n\n"])), function (pr) { return mixin(pr.size); }, function (pr) { return pr.Color || '#ff5722'; });
export var LoaderWrapper = styled.div(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n    ", "\n    display: ", ";\n"], ["\n    ", "\n    display: ", ";\n"])), function (pr) { return pr.disabled === true ? 'pointer-events: none;opacity: 0.4;' : ''; }, function (pr) { return pr.display ? 'inline-block' : 'none'; });
export var StyledWrapper = styled.div(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n    -webkit-box-flex: 0;\n    -ms-flex: 0 0 25%;\n    flex: 0 0 25%;\n    border: 1px solid rgba(255, 255, 255, 0.1);\n    -webkit-box-sizing: border-box;\n    box-sizing: border-box;\n    margin: 0;\n    position: relative;\n    display: -webkit-box;\n    display: -ms-flexbox;\n    display: flex;\n    -webkit-box-pack: center;\n    -ms-flex-pack: center;\n    justify-content: center;\n    -webkit-box-align: center;\n    -ms-flex-align: center;\n    align-items: center;\n    overflow: hidden;\n"], ["\n    -webkit-box-flex: 0;\n    -ms-flex: 0 0 25%;\n    flex: 0 0 25%;\n    border: 1px solid rgba(255, 255, 255, 0.1);\n    -webkit-box-sizing: border-box;\n    box-sizing: border-box;\n    margin: 0;\n    position: relative;\n    display: -webkit-box;\n    display: -ms-flexbox;\n    display: flex;\n    -webkit-box-pack: center;\n    -ms-flex-pack: center;\n    justify-content: center;\n    -webkit-box-align: center;\n    -ms-flex-align: center;\n    align-items: center;\n    overflow: hidden;\n"])));
var templateObject_1, templateObject_2, templateObject_3;
