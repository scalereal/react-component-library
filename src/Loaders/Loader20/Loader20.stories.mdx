import Loader20, { SizeMap } from '.';
import { Preview, Story, Meta, Props, ArgsTable, Canvas } from '@storybook/addon-docs/blocks';
import { withKnobs, color, boolean, select } from '@storybook/addon-knobs';

<Meta
    title="Components/Loaders/Loader20"
    component={Loader20}
    argTypes={{
        size: {
            control: {
                type: 'select',
                options: Object.values(SizeMap)
            }
        },
        Color: {
            control: 'color'
        },
        display: {
            control: {
                type: 'boolean'
            }
        },
        ref: {
            table: {
                disable: true
            }
        },
        theme: {
            table: {
                disable: true
            }
        },
        as: {
            table: {
                disable: true
            }
        }
    }}
/>

# Loaders

Displays a Loader.

export const Template = args => (
    <div>
        <Loader20 {...args} />
    </div>
);

<Preview>
    <Story name="Loader20" args={{ size: 'S' }}>
        {Template.bind({})}
    </Story>
</Preview>

### Color Variations

You can either change the theme or pass the colors as props.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-around', flexWrap: 'wrap' }}>
        <Loader20 size="S" Color="green" />
        <Loader20 size="S" Color="red" />
    </div>
</Preview>

### Sizes

There are four sizes available `S` | `M` | `L` | `XL`.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-around', flexDirection: 'column' }}>
        <Loader20 size="S" />
        <Loader20 size="M" />
        <Loader20 size="L" />
        <Loader20 size="XL" />
    </div>
</Preview>

### Props

<Props story="Loader20" />
