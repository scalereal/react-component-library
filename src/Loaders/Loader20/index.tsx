import * as React from 'react';
import { WithStyle } from '../../utils';
import { Props } from '../types';
import { Loader, LoaderWrapper, StyledWrapper } from './styled';

export const SizeMap: Props['size'][] = ['S', 'M', 'L', 'XL'];

const Loader20:React.FC<Props> & WithStyle = React.memo(React.forwardRef( (props, ref) => {
    const {size} = props;

    return (
        <LoaderWrapper ref={ref} size={size} {...props}>
            <StyledWrapper {...props}>
                <Loader {...props}>
                    <div className="multi-spinner">
                        <div className="multi-spinner">
                            <div className="multi-spinner">
                                <div className="multi-spinner">
                                    <div className="multi-spinner">
                                        <div className="multi-spinner" />
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                </Loader>
            </StyledWrapper>
        </LoaderWrapper>
    );
    
}));

Loader20.displayName="Loader20";
Loader20.defaultProps = {
    size:'S',
    visible: true
};

export default Loader20;
