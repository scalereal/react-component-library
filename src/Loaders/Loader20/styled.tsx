import styled from 'styled-components';
import { Props } from '../types';

const mixin = (size:any) => {
    switch (size) {
        case "S":
            return `
            height: 60px;
            width: 60px;
            .multi-spinner {
                border: 3px solid transparent;
            }
            `;
        
        case "M":
            return `
            height: 80px;
            width: 80px;
            .multi-spinner {
                border: 4px solid transparent;
            }
            `;

        case "L":
            return `
                height: 120px;
                width: 120px;
                .multi-spinner {
                    border: 5px solid transparent;
                }
            `;
        
        case "XL":
            return `
            height: 150px;
            width: 150px;
            .multi-spinner {
                border: 5px solid transparent;
            }
            `;
  
    }
 };

export const Loader = styled.div<Props>`
    ${pr=> mixin(pr.size)}

    position: relative;
    margin: auto;
    overflow: hidden;

    .multi-spinner {
        position: absolute;
        width: calc(100% - 9.9px);
        height: calc(100% - 9.9px);
        border-top-color: ${pr=>pr.Color || '#ff5722'};
        border-radius: 50%;
        -webkit-animation: spin 5s cubic-bezier(0.17, 0.49, 0.96, 0.76) infinite;
        animation: spin 5s cubic-bezier(0.17, 0.49, 0.96, 0.76) infinite;
    }

    @-webkit-keyframes spin {
        from {
            -webkit-transform: rotate(0deg);
            transform: rotate(0deg);
        }
        to {
            -webkit-transform: rotate(360deg);
            transform: rotate(360deg);
        }
    }

    @keyframes spin {
        from {
            -webkit-transform: rotate(0deg);
            transform: rotate(0deg);
        }
        to {
            -webkit-transform: rotate(360deg);
            transform: rotate(360deg);
        }
    }

`;

export const LoaderWrapper = styled.div<Props>`
    ${pr => pr.disabled === true?'pointer-events: none;opacity: 0.4;':''}
    display: ${pr=>pr.visible?'inline-block':'none'};
`;

export const StyledWrapper = styled.div<Props>`
    -webkit-box-flex: 0;
    -ms-flex: 0 0 25%;
    flex: 0 0 25%;
    border: 1px solid rgba(255, 255, 255, 0.1);
    -webkit-box-sizing: border-box;
    box-sizing: border-box;
    margin: 0;
    position: relative;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    overflow: hidden;
`;
