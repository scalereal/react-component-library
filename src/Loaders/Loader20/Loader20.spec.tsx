import { render } from '@testing-library/react';
import * as React from 'react';
import Loader20 from '.';

describe('Loader20 Loader', () => {
    it('should render with default theme', () => {
        const { container } = render(<Loader20/>);
        expect(container).toMatchSnapshot();
    });
    it('should render with all the props given', () => {
        const { container } = render(
            <Loader20
                size="M"
                Color='#126AFA'
            />
        );
        expect(container).toMatchSnapshot();
    });

    it('should render image avatar properly', () => {
        const { container } = render(
            <Loader20 size="L" />
        );
        expect(container).toMatchSnapshot();
    });

});