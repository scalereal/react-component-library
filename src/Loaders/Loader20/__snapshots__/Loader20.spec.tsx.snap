// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader20 Loader should render image avatar properly 1`] = `
<div>
  <div
    class="sc-gsTCUz hdkynM"
  >
    <div
      class="sc-dlfnbm bIJBxZ"
    >
      <div
        class="sc-bdfBwQ iBlNGP"
      >
        <div
          class="multi-spinner"
        >
          <div
            class="multi-spinner"
          >
            <div
              class="multi-spinner"
            >
              <div
                class="multi-spinner"
              >
                <div
                  class="multi-spinner"
                >
                  <div
                    class="multi-spinner"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Loader20 Loader should render with all the props given 1`] = `
<div>
  <div
    class="sc-gsTCUz hdkynM"
  >
    <div
      class="sc-dlfnbm bIJBxZ"
    >
      <div
        class="sc-bdfBwQ bGYsGj"
      >
        <div
          class="multi-spinner"
        >
          <div
            class="multi-spinner"
          >
            <div
              class="multi-spinner"
            >
              <div
                class="multi-spinner"
              >
                <div
                  class="multi-spinner"
                >
                  <div
                    class="multi-spinner"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Loader20 Loader should render with default theme 1`] = `
<div>
  <div
    class="sc-gsTCUz hdkynM"
  >
    <div
      class="sc-dlfnbm bIJBxZ"
    >
      <div
        class="sc-bdfBwQ iZnijR"
      >
        <div
          class="multi-spinner"
        >
          <div
            class="multi-spinner"
          >
            <div
              class="multi-spinner"
            >
              <div
                class="multi-spinner"
              >
                <div
                  class="multi-spinner"
                >
                  <div
                    class="multi-spinner"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
